frontend/components/text2text/results/Text2TextPredictions.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+  <div class="predictions">
+    <div
+      class="predictions__panel"
+      :class="arePredictionsVisible ? '--expanded' : '--collapsed'"
+    >
+      <div class="predictions__header" @click="toggleVisibility">
+        <base-button
+          v-if="!arePredictionsVisible"
+          class="predictions__button"
+          :class="arePredictionsVisible === 'Prediction' ? '--active' : null"
+          >Prediction</base-button
+        >
+        <svgicon
+          @click="arePredictionsVisible && toggleVisibility"
+          class="predictions__icon"
+          :name="arePredictionsVisible ? 'chevron-up' : 'chevron-down'"
+          width="16"
+          height="16"
+        />
+      </div>
+      <div v-if="arePredictionsVisible" class="predictions__content">
+        <div class="predictions__tabs">
+          <p class="predictions__title">Prediction:</p>
+          <div class="predictions__tabs__wrapper">
+            <base-button
+              :class="selectedPredictionIndex === index ? '--active' : null"
+              @click="selectPrediction(index)"
+              v-for="(prediction, index) in predictions"
+              :key="index"
+              data-title="Score"
+            >
+              {{ prediction.score | percent }}</base-button
+            >
+          </div>
+        </div>
+        <transition name="fade" appear :key="selectedPredictionIndex">
+          <div class="predictions__text">
+            {{ selectedPrediction }}
+          </div>
+        </transition>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import "assets/icons/chevron-down";
+import "assets/icons/chevron-up";
+import { IdState } from "vue-virtual-scroller";
+export default {
+  mixins: [
+    IdState({
+      // You can customize this
+      idProp: (vm) => `${vm.record.id}`,
+    }),
+  ],
+  props: {
+    record: {
+      type: Object,
+      required: true,
+    },
+    predictions: {
+      type: Array,
+      required: true,
+    },
+  },
+  idState() {
+    return {
+      arePredictionsVisible: false,
+      selectedPredictionIndex: 0,
+    };
+  },
+  computed: {
+    arePredictionsVisible: {
+      get: function () {
+        return this.idState.arePredictionsVisible;
+      },
+      set: async function (newValue) {
+        this.idState.arePredictionsVisible = newValue;
+      },
+    },
+    selectedPredictionIndex: {
+      get: function () {
+        return this.idState.selectedPredictionIndex;
+      },
+      set: async function (newValue) {
+        this.idState.selectedPredictionIndex = newValue;
+      },
+    },
```

Comment:
syntactic sugar => in an object, you can replace a function like 
`get : function() {...}` by `get() {...}`. Same with set